datajoint/connection.py

```diff
@@ -92,6 +92,16 @@ def connect(self):
         with warnings.catch_warnings():
             warnings.filterwarnings('ignore', '.*deprecated.*')
             self._conn = client.connect(
+                init_command=self.init_fun,
+                sql_mode="NO_ZERO_DATE,NO_ZERO_IN_DATE,ERROR_FOR_DIVISION_BY_ZERO,"
+                         "STRICT_ALL_TABLES,NO_ENGINE_SUBSTITUTION",
+                charset=config['connection.charset'],
+                **self.conn_info)
+
+            # If the database version is older then 8.0, then include 'NO_AUTO_CREATE_USER'
+            if self._conn.server_version > '8.0.0':
```

Comment:
```suggestion
            if self._conn.server_version < '8.0.0':
```

But should probably use version `distutil.version` for version comparisons. https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python 

Also, perhaps we should just drop `NO_AUTO_CREATE_USER` for all MySQL or move `sql_mode` into `dj.config`.

Reply:
For now, would you just remove `NO_AUTO_CREATE_USER` altogether?

Reply:
We could, but that will allow people using SQL server ver < 8.0 to create users via grants command, which is not a major issue since normally people don't create users via GRANT command. I just wanted to get feed back to see if there was anything that actually depends on that flag that I don't know about.

Reply:
Also thanks for flipping that '>' I was testing the second version to see if works.